plugins/kn-admin/cmd/domain/set.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright © 2019 NAME HERE <EMAIL ADDRESS>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package domain
+
+import (
+	"fmt"
+	"github.com/knative/client-contrib/plugins/kn-admin/cmd"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+)
+
+var selector []string
+var domain string
+
+// updateCmd represents the update command
+
+func NewDomainSetCommand(p *cmd.AdminParams) *cobra.Command {
+	domainSetCommand := &cobra.Command{
+		Use:   "set",
+		Short: "set route domain",
+		Long: `set Knative route domain for service
+
+For example:
+# To set a default route domain
+kn admin domain set --custom-domain mydomain.com
+# To set a route domain for service having label app=v1
+kn admin domain set --custom-domain mydomain.com --selector app=v1
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+			currentCm := &corev1.ConfigMap{}
+			currentCm, err := p.ClientSet.CoreV1().ConfigMaps("knative-serving").Get("config-domain", metav1.GetOptions{})
+			if err != nil {
+				fmt.Println("failed to get ConfigMaps:", err)
+				os.Exit(1)
+			}
+
+			desiredCm := currentCm.DeepCopy()
```

Comment:
Can we split this function into smaller private functions with clear name to what is going on :) thanks

Reply:
Similar to the `splitByEqualSign`. There are three immediate that come to mind.